analysis.R

```diff
@@ -0,0 +1,122 @@
+#this R file will be used to analyze the patient data
+# with the help of the functions in SupportingFunctions.R
+
+#setting current working directory; Change for your system
+setwd("/Users/kevinbuck/Desktop/Biocomputing/Exercises/Rproject")
+source("SupportingFunctions.R")
+####Initial Data Setup ####
+#turning all the txt files for Country Y into CSV files
+for (i in 120:175){
+  makeCSV(filename = paste("countryY/screen_",i,".txt",sep=""),delimeter = " ",
+          fileOutname = paste("countryY/screen_",i,sep=""))
+}
+
+#making giant data frame of screening data from both countries
+countryX.df <- combineCSVs(directoryname = "countryX/", country="X",na_handling = "remove",
+                           FileOutName = "CountryX")
+countryY.df <- combineCSVs(directoryname = "countryY/", country="Y",na_handling = "remove",
+                           FileOutName = "CountryY")
+
+allData.df <- rbind(countryX.df,countryY.df)
+
+#summarize data 
+SummarizeCompiledTests(allData.df)
+
+
+#making function to check if the screened patient is infected
+check_infected <- function(row_index,df){
+  return(sum(as.numeric(df[row_index,3:12] == "1")) != 0)
+}
+
+####Tracking Cases Over Time####
+
+day.list <- as.character(seq(120,175))
+
+#initiating data frame to store info
+Daily_Total_Cases <- data.frame(Day=as.numeric(day.list),CountryX.Total=numeric(length(day.list)),
+                                  CountryY.Total=numeric(length(day.list)))
+row.names(Daily_Total_Cases) = day.list
+#this big loop checks if a screened row has any markers present (is patient infected?)
+# and then it will add 1 to the total for cases in the appropriate day as stored in the dataframe
+for(day_num in day.list){
+  tempDF <- allData.df[allData.df[,"dayofYear"] == day_num,]
+  for(i in 1:nrow(tempDF)){
+    if( (check_infected(i,tempDF) == TRUE) & (tempDF[i,'country'] == "X")){
+      Daily_Total_Cases[day_num,"CountryX.Total"] = 1 + Daily_Total_Cases[day_num,"CountryX.Total"]
+    }
+    if((check_infected(i,tempDF) == TRUE) & (tempDF[i,'country'] == "Y")){
+      Daily_Total_Cases[day_num,"CountryY.Total"] = 1 + Daily_Total_Cases[day_num,"CountryY.Total"]
+    }
+  }
+}
+
+#####Plotting Daily Cases by Day#####
+
+#Notice how Country X has cases starting at day 120, and the country Y
+# cases line is at 0 cases until day 138ish. This means the outbreak began in country X
+
+library(ggplot2)
+ggplot()+geom_line(data=Daily_Total_Cases,mapping=aes(x=Day,y=CountryX.Total,color="CountryX")) +
+  geom_line(data=Daily_Total_Cases,mapping=aes(x=Day,y=CountryY.Total,color="CountryY")) +
+  ylab("Daily Cases") + xlab("Day Number") + ggtitle("Infections Over Time")+ 
+  scale_color_manual(name = "Country", values = c("CountryX" = "blue", "CountryY" = "red")) 
+
```

Comment:
For more concise code:
ggplot(data=Daily_Total_Cases,mapping=aes(x=Day))+
geom_line(mapping=aes(y=CountryX.Total,color="CountryX")) +
  geom_line(mapping=aes(y=CountryY.Total,color="CountryY")) +
  ylab("Daily Cases") + xlab("Day Number") + ggtitle("Infections Over Time")+ 
  scale_color_manual(name = "Country", values = c("CountryX" = "blue", "CountryY" = "red"))